core/src/main/java/org/apache/calcite/rel/metadata/RelMdColumnUniqueness.java

```diff
@@ -348,11 +361,15 @@ public Boolean areColumnsUnique(Intersect rel, RelMetadataQuery mq,
       return leftJoinColsUnique && rightUnique;
     }
 
-    throw new AssertionError();
```

Comment:
The constraint that `columns.cardinality() == 0` is removed at the beginning of this method.